kyaml/kio/pkgio_writer.go

```diff
@@ -65,45 +71,36 @@ func (r LocalPackageWriter) Write(nodes []*yaml.RNode) error {
 	// validate outputs before writing any
 	for path := range outputFiles {
 		outputPath := filepath.Join(r.PackagePath, path)
-		if st, err := os.Stat(outputPath); !os.IsNotExist(err) {
-			if err != nil {
-				return errors.Wrap(err)
-			}
-			if st.IsDir() {
-				return fmt.Errorf("config.kubernetes.io/path cannot be a directory: %s", path)
-			}
+		if r.FileSystem.IsDir(outputPath) {
+			return fmt.Errorf("config.kubernetes.io/path cannot be a directory: %s", path)
 		}
 
-		err = os.MkdirAll(filepath.Dir(outputPath), 0700)
+		err = r.FileSystem.MkdirAll(filepath.Dir(outputPath))
 		if err != nil {
 			return errors.Wrap(err)
 		}
 	}
 
 	// write files
+	buf := bytes.NewBuffer(nil)
 	for path := range outputFiles {
 		outputPath := filepath.Join(r.PackagePath, path)
-		err = os.MkdirAll(filepath.Dir(filepath.Join(r.PackagePath, path)), 0700)
+		err = r.FileSystem.MkdirAll(filepath.Dir(filepath.Join(r.PackagePath, path)))
 		if err != nil {
 			return errors.Wrap(err)
 		}
 
-		f, err := os.OpenFile(outputPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, os.FileMode(0600))
-		if err != nil {
+		buf.Reset()
+		w := ByteWriter{
+			Writer:                buf,
```

Comment:
Rather than writing to a buffer and then to the fs, would it work to do `r.FileSystem.Create(path)` and pass the result to the `ByteWriter` here, more similarly to what was being done before?

Reply:
The reason is that there's no easy way with the current filesystem abstraction to open a file with O_TRUNC